apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,999 @@
+	if (ret < 0) {
+		PRNT("ERROR: setsockopt failed %d", ret);
+		return WS_TEST_SOCKET_CONNECT_ERR;
+	}
+
+	/* Get the server address from the host name */
+	server.sin_family = AF_INET;
+	server.sin_port = htons(ws->port);
+	ret = ws_test_gethostip(ws->hostname, &server.sin_addr.s_addr);
+	if (ret < 0) {
+		/* Could not resolve host (or malformed IP address) */
+		PRNT("ERROR: Failed to resolve hostname %d", ret);
+		close(sockfd);
+		return WS_TEST_SOCKET_CONNECT_ERR;
+	}
+
+	/* Connect to server.  First we have to set some fields in the
+	 * 'server' address structure.  The system will assign me an arbitrary
+	 * local port that is not in use.
+	 */
+	ret = connect(sockfd, (struct sockaddr *)&server, sizeof(struct sockaddr_in));
+	if (ret < 0) {
+		PRNT("ERROR: connect failed: %d errno: %d", ret, errno);
+		close(sockfd);
+		return WS_TEST_SOCKET_CONNECT_ERR;
+	}
+	FUNC_EX;
+	return sockfd;
+}
+
+#ifdef CONFIG_NET_SECURITY_TLS
+void ws_test_chunk_transfer(struct http_client_request_t *param, 
+							int send_len, int sockfd, 
+							struct http_client_tls_t *client_tls) {
+	FUNC_EN;
+	int ret, cbuf, chunks = param->chunk_count - 1;
+	//send headers
+	int buf_len = 0;
+	int sndlen = send_len-g_last_clen;
+	while (sndlen > 0) {
+		if (param->tls) {
+			ret = mbedtls_ssl_write(&(client_tls->tls_ssl),
+									(const unsigned char *)param->buffer + buf_len,
+									sndlen);
+		} else {
+			ret = send(sockfd, param->buffer + buf_len, sndlen, 0);
+		}
+		if (ret < 1) {
+			PRNT("ERROR: send failed: %d", ret);
+			goto errout;
+		} else {
+			sndlen -= ret;
+			buf_len += ret;
+			PRNT("SEND SUCCESS: send %d bytes", ret);
+		}
+	}
+	//send chunks
+	cbuf = buf_len;
+	while(chunks--) {
+		buf_len = cbuf;
+		sndlen = g_normal_clen;
+		while (sndlen > 0) {
+			if (param->tls) {
+				ret = mbedtls_ssl_write(&(client_tls->tls_ssl),
+										(const unsigned char *)param->buffer + buf_len,
+										sndlen);
+			} else {
+				ret = send(sockfd, param->buffer + buf_len, sndlen, 0);
+			}
+			if (ret < 1) {
+				PRNT("ERROR: send failed: %d", ret);
+				goto errout;
+			} else {
+				sndlen -= ret;
+				buf_len += ret;
+				PRNT("SEND SUCCESS: send %d bytes", ret);
+			}
+		}
+	}
+	//last chunk
+	buf_len = cbuf;
+	sndlen = g_last_clen;
+	while (sndlen > 0) {
+		if (param->tls) {
+			ret = mbedtls_ssl_write(&(client_tls->tls_ssl),
+									(const unsigned char *)param->buffer + buf_len,
+									sndlen);
+		} else {
+			ret = send(sockfd, param->buffer + buf_len, sndlen, 0);
+		}
+		if (ret < 1) {
+			PRNT("ERROR: send failed: %d", ret);
+			goto errout;
+		} else {
+			sndlen -= ret;
+			buf_len += ret;
+			PRNT("SEND SUCCESS: send %d bytes", ret);
+		}
+	}
+
+errout:
+	if (param->callback && param->response) {
+		ws_test_http_client_response_release(param->response);
+	}
+	if (param->tls) {
+		ws_test_tls_ssl_release(client_tls);
+	}
+	FUNC_EX;
+}
+
+int ws_test_parsehttpurl(const char *url, uint16_t *port, char *hostname, int hostlen) {
```

Comment:
How about to move this function to webserver_test_util.c. The function seems to have utility characteristic.